lib/contentful_model/manageable.rb

```diff
@@ -0,0 +1,104 @@
+module ContentfulModel
+  module Manageable
+    attr_reader :dirty
+
+    def initialize(*args)
+      @dirty = false
+      define_setters
+    end
+
+    def to_management
+      update_management_entry_fields(management_entry)
+    end
+
+    def save
+      result = to_management.save
+      @dirty = false
+      self
+    end
+
+    def publish
+      to_management.publish
+      self
+    end
+
+    protected
+
+    def update_management_entry_fields(management_entry)
+      map_management_fields(management_entry, fields)
+    end
+
+    private
+
+    def management_space
+      @management_space ||= self.class.management(default_locale: locale).spaces.find(space.id)
+    end
+
+    def management_entry
+      # Fetches always in case of Version changes to avoid Validation errors
```

Comment:
I'm aware of it being a bit aggressive, but couldn't find a middle-ground approach, if I cache the management entry, and update data after it's been fetched, we can run into race conditions and receive validation errors upon save.

So yes, local data wins always, entries should have local data updated until the last fetch, and then it's whatever the user sent that will be sent ultimately.

If you have any other possible approach?

Having a list of modified fields and using those only to overwrite is what comes to mind, the problem is that somehow management entries come with `fields` empty, so you need to populate them anyway in order to save. I think that's avoidable using the new Content Type Cache in CMA, but it's something I haven't yet checked.